google/cloud/storage/client.cc

```diff
@@ -96,6 +97,9 @@ std::shared_ptr<internal::RawClient> Client::CreateDefaultInternalClient(
       rest_opts.set<rest::EnableCurlSigpipeHandlerOption>(
           opts.get<EnableCurlSigpipeHandlerOption>());
     }
+    rest_opts.set<rest::IgnoredHttpErrorCodes>(
```

Comment:
Does this means the status codes get ignored for all requests?  How does it work for other requests then?  Maybe add a comment?